transport/tchannel/channel_outbound.go

```diff
@@ -253,3 +257,13 @@ func getResponseErrorAndDeleteHeaderKeys(headers transport.Headers) error {
 	errorMessage, _ := headers.Get(ErrorMessageHeaderKey)
 	return intyarpcerrors.NewWithNamef(errorCode, errorName, errorMessage)
 }
+
+func checkServiceMatchAndDeleteHeaderKey(reqSvcName string, resHeaders transport.Headers) (bool, string) {
+	defer func() {
+		resHeaders.Del(ServiceHeaderKey)
+	}()
+	if resSvcName, ok := resHeaders.Get(ServiceHeaderKey); ok {
+		return reqSvcName == resSvcName, resSvcName
```

Comment:
Since we know whether we get the service key in this function already, let's remove the `defer` above and move the deletion just above this return.
```go
if resSvcName, ok := resHeaders.Get(ServiceHeaderKey); ok {
 resHeaders.Del(ServiceHeaderKey)
 return reqSvcName == resSvcName, resSvcName
}
```

Reply:
Defer statement is removed.